db-ops/job-create-db-mysql.yml.tpl

```diff
@@ -0,0 +1,114 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: db-ops
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: db-create-users
+  namespace: db-ops
+type: Opaque
+stringData: 
+  create-user.sql: |
+    CREATE USER '$DB_APP_USERNAME' IDENTIFIED BY '$DB_APP_PASSWORD';
+    GRANT ALL PRIVILEGES ON $DB_NAME.* TO '$DB_APP_USERNAME';
+  RDS_MASTER_PASSWORD: $SECRET_PASSWORD
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: $PROJECT_NAME
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: $PROJECT_NAME
+  namespace: $PROJECT_NAME
+type: Opaque
+stringData: 
+  DATABASE_USERNAME: $DB_APP_USERNAME
+  DATABASE_PASSWORD: $DB_APP_PASSWORD
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: db-create-users
+  namespace: db-ops
+spec:
+  template:
+    spec:
+      containers:
+      - name: create-rds-user
+        image: $DOCKER_IMAGE_TAG
+        command: 
+        - sh
+        args: 
+        - '-c' 
+        - mysql -u$MASTER_RDS_USERNAME -h $DB_ENDPOINT $DB_NAME < /db-ops/create-user.sql
+        env:
+        - name: DB_ENDPOINT
+          value: $DB_ENDPOINT
+        - name: DB_NAME
+          value: $DB_NAME
+        - name: MYSQL_PWD
+          valueFrom:
+            secretKeyRef:
+              name: db-create-users
+              key: RDS_MASTER_PASSWORD
+        volumeMounts:
+        - mountPath: /db-ops/create-user.sql
+          name: db-create-users
+          subPath: create-user.sql
+      volumes:
+        - name: db-create-users
+          secret:
+            secretName: db-create-users
+      restartPolicy: Never
+  backoffLimit: 1
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: db-pod
+  namespace: $PROJECT_NAME
+spec:
+# this is purposely left at 0 so it can be enabled for troubleshooting purposes
+  replicas: 0
+  selector:
+    matchLabels:
+      app: db-pod
+  template:
+    metadata:
+      labels:
+        app: db-pod
+    spec:
+      automountServiceAccountToken: false
+      containers:
+      - command:
+        - sh
+        args: 
+        - "-c"
+        # long running task so the pod doesn't exit with 0
+        - tail -f /dev/null
+        image: $DOCKER_IMAGE_TAG
+        imagePullPolicy: Always
+        name: db-pod
+        env:
+        - name: DB_ENDPOINT
+          value: $DB_ENDPOINT
+        - name: DB_NAME
+          value: $DB_NAME
+        - name: DB_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: $PROJECT_NAME
+              key: DATABASE_PASSWORD
+        volumeMounts:
+        - mountPath: /db-ops/rds-password
```

Comment:
Why are we mounting it in addition to exposing it as an env var?

Reply:
ahh sorry forgot to remove it.
I was running into some issues with MySQL 8.0 
- in 8.0 mysql deprecated using the env-var `mysql_pwd`, should we also avoid using so? 
- 8.0 defaults to `caching_sha2_password` as their password storing which the alpine mysql-client doesn't support, I changed my rds to 5.7 to test this, since I couldn't find a work around that let me connect

do you think it's reasonable to use MySQL 5.7?

Reply:
If we have the env vars can't we just use them directly in the cli though? like `mysql -u <user> -p<password>`

Reply:
it solves the deprecated `mysql_pwd` env var, but not the the `caching_sha2_password`, it just doesn't connect unless its stored in the database differently from creation time it seems, most solutions includes editing the database's `my.cnf` it seems, or while creating specify a different method https://stackoverflow.com/a/49935803/1753765